server/api/users.js

```diff
@@ -1054,23 +1056,37 @@ router.put('/users/:id/civic/trial', jwtAuth('write'), async (req, res, next) =>
       return;
     }
 
-    const doc = await dbRef.doc(id).get();
-    if (!doc.exists) throw new Error('USER_NOT_EXIST');
+    const userDoc = await dbRef.doc(id).get();
+    if (!userDoc.exists) throw new Error('USER_NOT_EXIST');
+
+    const configDoc = await configRef.doc('civicLiker').get();
+    const preRegQuota = (configDoc.exists && configDoc.data().preRegQuota) || 0;
 
-    const updateObj = {
-      isPreRegCivicLiker: true,
-    };
+    const preRegQuery = await dbRef
+      .where('isPreRegCivicLiker', '==', true)
+      .limit(preRegQuota)
+      .get();
```

Comment:
should update a `total count` value when there is a new civicliker, instead of query whole db for every read

Reply:
Now minus quota for each new Civic Liker

Reply:
Updated with total count